WordPress/src/main/java/org/wordpress/android/networking/GravatarApi.java

```diff
@@ -76,17 +81,41 @@ public void run() {
                                 if (response.isSuccessful()) {
                                     gravatarUploadListener.onSuccess();
                                 } else {
+                                    Map<String, Object> properties = new HashMap<>();
+                                    properties.put("network_response_code", response.code());
+
+                                    String responseBody;
+                                    try {
+                                        responseBody = response.body().string();
+                                    } catch (IOException e) {
+                                        responseBody = "null";
+                                    }
+                                    properties.put("network_response_body", responseBody);
+
+                                    AnalyticsTracker.track(AnalyticsTracker.Stat.ME_GRAVATAR_UPLOAD_UNSUCCESSFUL,
+                                            properties);
+                                    AppLog.w(AppLog.T.API, "Network call unsuccessful trying to upload Gravatar: " +
+                                            responseBody);
                                     gravatarUploadListener.onError();
                                 }
                             }
                         });
                     }
 
                     @Override
-                    public void onFailure(okhttp3.Call call, IOException e) {
+                    public void onFailure(okhttp3.Call call, final IOException e) {
                         new Handler(Looper.getMainLooper()).post(new Runnable() {
                             @Override
                             public void run() {
+                                Map<String, Object> properties = new HashMap<>();
+                                properties.put("network_exception_class", e != null ? e.getClass().getCanonicalName()
+                                        : "null");
+                                properties.put("network_exception_message", e != null ? e.getMessage() : "null");
+                                AnalyticsTracker.track(AnalyticsTracker.Stat.ME_GRAVATAR_UPLOAD_EXCEPTION, properties);
+                                CrashlyticsUtils.logException(e, CrashlyticsUtils.ExceptionType.SPECIFIC,
+                                        AppLog.T.API, "Network call failure trying to upload Gravatar!");
+                                AppLog.w(AppLog.T.API, "Network call failure trying to upload Gravatar!" + e
```

Comment:
Oops, yeah, missed that one from the treatment the previous lines got. Good catch!